Vagrantfile

```diff
@@ -111,8 +111,8 @@ Vagrant.configure("2") do |config|
     sudo consul members || true
     nomad server members || true
     nomad status || true
-    vault version || true
-    vault init || true
+    echo "$(vault version)" || true
```

Comment:
Hey, Paul. Just curious: why change `vault version` to `echo "$(vault version)"`? Aren't they the same, and the second one is just extra complexity? Or am I missing something here?

Reply:
@neilmayhew  I didn't test in a Linux terminal but in a macos it was showing the version with  new line`\n` for each caracter like I added in the issue https://github.com/fpco/fpco-salt-formula/issues/148

Reply:
Thanks for explaining. I get that too, even in a pure Linux setup. I think it must be to do with how Vagrant is buffering terminal output. For example, I get things like this sometimes:

```
    default: Error initializi
    default: ng: Put https:/
    default: /127.0.0.1:8200/v1/sys/init: dial tcp 127.0.0.1:8200: connect: co
    default: nnection refused
```

I assume Vagrant is polling for output and printing it as it arrives. If vault's output is slow, Vagrant will see it arriving one character at a time, leading to the effect we're seeing from the `vault version` command. By collecting the `version` output into a string, it's re-output by `echo` as a whole, and I assume `echo` is fast enough to be able to get all of the output sent to Vagrant in one piece.

However, since I'm seeing a similar effect with other commands, this seems to be a more general problem. I don't know what a general solution might look like, though.

Reply:
Yeah, is how vagrant is buffering terminal output because they aren't controlling the data that is being streamed from the VM so not sure if right now there is a general solution for this kind of issues. Here vagrant is adding a line for each buffering https://github.com/hashicorp/vagrant/blob/9c299a2a357fcf87f356bb9d56e18a037a53d138/plugins/provisioners/shell/provisioner.rb#L37

Reply:
But I think if we make that the VM generate the string as a suboperation `echo "$(command)"` we can get the output in one line.

Reply:
I checked with:

```
strace -f -e trace=write vault version >/dev/null
```

and, sure enough, `vault` is writing to stdout one character at a time. When output is a tty, however, it's line-buffered. This is a "feature" of Go, [apparently](https://grokbase.com/t/gg/golang-nuts/158cgb3rfd/go-nuts-os-stdout-is-not-buffered). Seems backwards to me!

You can also work around this by piping to `cat`, which maybe is a little more readable:

```
vault version | cat
```

Reply:
@neilmayhew yeah, I like `vault version | cat` seems simpler  👍